pkg/config/config_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright © 2022 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+	"gotest.tools/v3/assert"
+)
+
+func TestBootstrapConfig(t *testing.T) {
```

Comment:
What about cases for no-config files or errors reading / writing?

Reply:
Thanks, addressed.